src/qt/transactionview.cpp

```diff
@@ -663,3 +665,14 @@ void TransactionView::closeOpenedDialogs()
     }
     m_opened_dialogs.clear();
 }
+
+bool TransactionView::detailsAlreadyShown(const QModelIndex &idx)
+{
+    for (TransactionDescDialog* dlg : m_opened_dialogs) {
+        if (dlg->getTransactionId() == idx.data(TransactionTableModel::TxHashRole).toString()) {
+            dlg->activateWindow();
```

Comment:
The dialog is not brought to the foreground in my case (I am using Mac). In order to make it work I had to call `raise()` instead. Thoughts?

```suggestion
            dlg->raise()
```

Reply:
Thanks for reviewing and spotting that. I tried on Ubuntu to call **only** `dlg->raise()` but doesn't seem to work (also try other things like `dlg->setFocus()`, `dlg->setFocusPolicy(Qt::StrongFocus)`, `dlg->focusWidget()` and some combinations), until I found this in the QT documentation about [activateWindow()](https://doc.qt.io/qt-5/qwidget.html#activateWindow) (which I should have checked first):

_This function performs the same operation as clicking the mouse on the title bar of a top-level window. On **X11**, the result depends on the Window Manager. **If you want to ensure that the window is stacked on top as well you should also call [raise](https://doc.qt.io/qt-5/qwidget.html#raise)**. Note that the window must be visible, otherwise activateWindow() has no effect._

_On **Windows**, if you are calling this when the application is not currently the active one then it will not make it the active window. It will change the color of the taskbar entry to indicate that the window has changed in some way. This is because Microsoft does not allow an application to interrupt what the user is currently doing in another application._

So, since you tried it on Mac already, let me take a look on how it behaves on **Windows**.

Reply:
In Mac, it seems that `activateWindow()` only makes the window active, i.e: the title of the dialog gets highlighted and gets the keyboard input focus. The `raise()` function brings the dialog on top. So, at least in Mac, it makes sense to keep both IMO

Reply:
Yes, it needs both `activateWindow()` & `raise()`, let me try it on Windows and I'll incorporate your [suggestion](https://github.com/bitcoin-core/gui/pull/817#discussion_r1569483043), thanks!